src/com/facebook/buck/swift/SwiftRuntimeNativeLinkable.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2016-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.swift;
+
+import static com.facebook.buck.swift.SwiftDescriptions.isSharedRequested;
+
+import com.facebook.buck.apple.AppleCxxPlatform;
+import com.facebook.buck.cxx.CxxPlatform;
+import com.facebook.buck.cxx.Linker;
+import com.facebook.buck.cxx.NativeLinkable;
+import com.facebook.buck.cxx.NativeLinkableInput;
+import com.facebook.buck.model.BuildTarget;
+import com.facebook.buck.model.Flavor;
+import com.facebook.buck.model.ImmutableFlavor;
+import com.facebook.buck.parser.NoSuchBuildTargetException;
+import com.facebook.buck.rules.SourcePath;
+import com.facebook.buck.rules.args.StringArg;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.nio.file.Path;
+
+/**
+ * Pseudo linkable for representing Swift runtime library's linker arguments.
+ */
+final class SwiftRuntimeNativeLinkable implements NativeLinkable {
```

Comment:
The problem I [mentioned above](https://github.com/facebook/buck/pull/862#discussion_r77158301) doesn't seem to work with flags. By using flags, I will have to put those flags inside `SwiftLibrary#getNativeLinkableInput`, and this will cause duplication.

Let take an example, we have an `apple_binary` A depends on `apple_library` B and another `swift_library` C which B depends on. A has another `swift_library` D. We want to create a binary for A with dynamic linking to all its dependencies (`link_style=shared`). We would ended up with duplication of swift runtime flags like this:
![Uploading image.png…]()